corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -51,6 +51,23 @@ function hqDefine(path, dependencies, moduleAccessor) {
 
     (function (factory) {
         if (typeof define === 'function' && define.amd && window.USE_REQUIREJS) {
+
+            // Before running build_requirejs, define doesn't know how to handle modules tagged with `es6!`
+            // After requirejs is built on production, es6 modules are always referenced without the es6! prefix
+            // when included as a dependency. However, when running locally without build_requirejs, it
+            // is essential to reference the es6 module with the es6! prefix.
+            if (window.IS_LOCAL_REQUIREJS_ES6) {
+                var es6Dependencies = [
+                    'hqwebapp/js/bootstrap5_loader',
+                ];
+                for (var i = 0; i < dependencies.length; i++) {
+                    var dependency = dependencies[i];
+                    if (es6Dependencies.indexOf(dependency) >= 0) {
```

Comment:
This is O(N^2). Not sure if the list will ever get big enough for that to be an issue? Would it be more efficient to make `es6Dependencies` an object and use `es6Dependencies.hasOwnProperty(dependency)` instead?